pandas/tests/indexing/test_multiindex.py

```diff
@@ -158,12 +158,44 @@ def test_loc_getitem_series(self):
         result = x.loc[[1, 3]]
         tm.assert_series_equal(result, expected)
 
+        # GH15424
+        y1 = Series([1, 3], index=[1, 2])
+        result = x.loc[y1]
+        tm.assert_series_equal(result, expected)
+
         empty = Series(data=[], dtype=np.float64)
         expected = Series([], index=MultiIndex(
             levels=index.levels, labels=[[], []], dtype=np.float64))
         result = x.loc[empty]
         tm.assert_series_equal(result, expected)
 
+    def test_loc_getitem_dataframe(self):
+        # passing a dataframe as a key with a MultiIndex
+        index = MultiIndex.from_product([[1, 2, 3], ['A', 'B', 'C']])
+        x = Series(index=index, data=range(9), dtype=np.float64)
+        idx_keys = [(1, 'A'), (2, 'C'), (3, 'B')]
```

Comment:
Lost again... aren't you referring to something like [this](https://github.com/pandas-dev/pandas/pull/15425/files/c516c4521719be34b134f4550f5f7b110e3902d7#diff-8ba06f84d19b53a58f7200f8fb502dc3R166) and [this](https://github.com/pandas-dev/pandas/pull/15425/files/c516c4521719be34b134f4550f5f7b110e3902d7#diff-8ba06f84d19b53a58f7200f8fb502dc3R193)?!